lightning/src/chain/keysinterface.rs

```diff
@@ -1047,6 +1054,15 @@ impl KeysInterface for KeysManager {
 	fn read_chan_signer(&self, reader: &[u8]) -> Result<Self::Signer, DecodeError> {
 		InMemorySigner::read(&mut std::io::Cursor::new(reader))
 	}
+
+	fn sign_invoice(&self, invoice_preimage: Vec<u8>) -> Result<RecoverableSignature, ()> {
+		let secp_ctx = Secp256k1::new();
+		let mut raw_hash: [u8; 32] = Default::default();
+		raw_hash.copy_from_slice(&Sha256::hash(&invoice_preimage)[..]);
+		let msg_hash = Message::from_slice(&raw_hash[..])
+			.expect("Hash is 32 bytes long, same as MESSAGE_SIZE");
+		Ok(secp_ctx.sign_recoverable(&msg_hash, &self.get_node_secret()))
```

Comment:
You can simplify this a good bit (plus then drop the `Message` import above.
```suggestion
		Ok(self.secp_ctx.sign_recoverable(&hash_to_message!(&Sha256::hash(&invoice_preimage)), &self.get_node_secret()))
```